server.js

```diff
@@ -0,0 +1,87 @@
+/**
+ *
+ * This application transforms socket messages into MQTT messages
+ *
+ * @author  Dennis de Greef <github@link0.net>
+ * @license MIT
+ *
+ * Requires JSON structure for each request according to the following format (on a single line)
+ *
+ * {
+ *     secret: "youshouldchangethis",
+ *     topic: "/user/example/foo",
+ *     message: "This is a message for the queue"
+ * }
+ *
+ * in which you'll get a response with a boolean whether the request was accepted
+ *
+ * {
+ *     success: true
+ * }
+ *
+ */
+var mqtt     = require('mqtt');
+var net      = require('net');
+var config   = require('./config');
+
+var mqttUri  = 'mqtt://' + config.mqtt.hostname + ':' + config.mqtt.port;
+var client   = mqtt.connect(mqttUri);
+
+client.on('connect', function () {
+    log("Connected to MQTT: " + mqttUri);
+    log("Starting to listen on tcp://" + config.socket.host + ':' + config.socket.port);
+    net.createServer(function(sock) {
+        sock.on('data', function(streamData) {
+            var data = streamData.toString();
+
+            var json = validateJSON(data.replace(/^\s+|\s+$/g, ''));
+            if (json && json.secret == config.socket.secret) {
```

Comment:
You are right, with Javascript you tend to go as deeply nested as possible for some reason :weary: 

I agree with your feedback. The point about security impact I even disagree, since it will only show up in the logs. I always return `success: false` to the socket. But great that you're also looking at it from a security perspective! :+1:

Thanks!